src/main/scala/ai/privado/cache/AuditCache.scala

```diff
@@ -0,0 +1,190 @@
+package ai.privado.cache
+
+import ai.privado.dataflow.DuplicateFlowProcessor
+import ai.privado.entrypoint.PrivadoInput
+import ai.privado.model.{Constants, DataFlowPathModel, InternalTag}
+import io.joern.dataflowengineoss.language.Path
+import io.shiftleft.semanticcpg.language._
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.util.{Failure, Success}
+
+object AuditCache {
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  private val flowPathBeforeFirstFiltering: mutable.Set[SourcePathInfo] = new mutable.HashSet[SourcePathInfo]()
+
+  private val flowPathBeforeSecondFiltering: mutable.Set[SourcePathInfo] = new mutable.HashSet[SourcePathInfo]()
+
+  private val flowPathBeforeFirstDedup: mutable.Set[SourcePathInfo] = new mutable.HashSet[SourcePathInfo]()
+
+  private val flowPathBeforeSecondDedup: mutable.Set[SourcePathInfo] = new mutable.HashSet[SourcePathInfo]()
+
+  private val flowPathFinal: mutable.Set[SourcePathInfo] = new mutable.HashSet[SourcePathInfo]()
+
+  private var dataflowMapByPathId: Map[String, Path] = null
+
+  def getFlowBeforeFirstFiltering: Set[SourcePathInfo] = flowPathBeforeFirstFiltering.toSet
+
+  def addIntoBeforeFirstFiltering(sourcePathInfo: SourcePathInfo): Unit = flowPathBeforeFirstFiltering += sourcePathInfo
+
+  def addIntoBeforeSecondFiltering(sourcePathInfo: SourcePathInfo): Unit =
+    flowPathBeforeSecondFiltering += sourcePathInfo
+
+  def checkFlowExistInSecondFiltering(sourcePathInfo: SourcePathInfo): Boolean =
+    flowPathBeforeSecondFiltering.contains(sourcePathInfo)
+
+  def pathIdExist(pathId: String): Boolean =
+    if (dataflowMapByPathId != null && dataflowMapByPathId.contains(pathId)) true else false
+
+  def getPathFromId(pathId: String): Path = dataflowMapByPathId(pathId)
+
+  def addIntoBeforeFirstFiltering(
```

Comment:
@ankit-privado You have repeated the same function which we have used in Dataflow to get to a discrete item, this is not a good practice as tomorrow is anything changes in those function these will get impacted.

Reply:
Actually, I have to change too many functions associated to main scan run that why i thought make it separate. Okay, i will change this too.